### PR TITLE
chore: add cache info to chart results

### PR DESCRIPTION
**Pull request by magnew**
Relates to: #7717 

### Description:

Adds cache information to chart results. When we hit the cache we return the time the cache was updated. This will be used to inform users. 

I tested this a fair amount, but a lot of the system passes through these functions. Is there anything I should be sure to do before merging this? 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 1]
```diff
@@ -1024,7 +1026,7 @@ export class ProjectService {
     async getResultsForChart(
         user: SessionUser,
         chartUuid: string,
-    ): Promise<Record<string, any>[]> {
+    ): Promise<{ rows: Record<string, any>[]; cacheMetadata?: CacheMetadata }> {
```
Comment (ZeRego) [inline packages/backend/src/services/ProjectService/ProjectService.ts:1029]: any reason for cacheMetadata to be optional ?

Comment (magnew) [reply packages/backend/src/services/ProjectService/ProjectService.ts:1029]: Nope. Will fix.

**Comment by magnew** [comment]
Unit tests are (unsurprisingly) failing. Will fix.

**Review by ZeRego — APPROVED** [review]

**Comment by magnew** [comment]
Discussed with @ZeRego -- we're going to merge this with a couple failing tests that are failing everywhere.

**Pull request merged**